File: translations/codenet/uncommented/C++2J/gpt-4o/Code/s194793609.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();

        long[] a = new long[n];
        for (int i = 0; i < n; i++) {
            a[i] = scanner.nextLong();
        }

        long[] abs_a = new long[n];
        for (int i = 0; i < n; i++) {
            abs_a[i] = Math.abs(a[i]);
        }

        long ans = 0;
        long min = abs_a[0];
        for (long e : abs_a) {
            ans += e;
            if (min > e) {
                min = e;
            }
        }

        int num_negative = 0;
        for (long e : a) {
            if (e < 0) {
                num_negative++;
            }
        }

        if (num_negative % 2 != 0) {
            ans -= min * 2;
        }

        System.out.println(ans);
        scanner.close();
    }
}

// <END-OF-CODE>

File: translations/codenet/uncommented/C++2P/gpt-4o/Code/s194793609.py
n = int(input())

a = list(map(int, input().split()))

abs_a = [abs(x) for x in a]

ans = 0
min_val = abs_a[0]
for e in abs_a:
    ans += e
    if min_val > e:
        min_val = e

num_negative = sum(1 for e in a if e < 0)

if num_negative % 2:
    ans -= min_val * 2

print(ans)

# <END-OF-CODE>
